src/main/java/org/opentripplanner/street/model/edge/StairsEdge.java

```diff
@@ -0,0 +1,154 @@
+package org.opentripplanner.street.model.edge;
+
+import org.locationtech.jts.geom.LineString;
+import org.opentripplanner.framework.geometry.CompactLineStringUtils;
+import org.opentripplanner.framework.i18n.I18NString;
+import org.opentripplanner.routing.api.request.preference.RoutingPreferences;
+import org.opentripplanner.street.model.StreetTraversalPermission;
+import org.opentripplanner.street.model.TurnRestriction;
+import org.opentripplanner.street.model.vertex.StreetVertex;
+import org.opentripplanner.street.search.state.State;
+
+/**
+ * Represents a flight of stairs that are derived from OSM data.
+ */
+public class StairsEdge extends OsmEdge {
+
+  // stairs are only allowed for walking and cycling/scootering
+  private static final StreetTraversalPermission DEFAULT_PERMISSIONS =
+    StreetTraversalPermission.PEDESTRIAN_AND_BICYCLE;
+
+  private final I18NString name;
+  private final byte[] compactGeometry;
+  private final double distance;
+  private float walkSafetyFactor = 1f;
```

Comment:
I don't think we should consider safety for stairs as we have a separate reluctance for stairs.